loki/ir/nodes.py

```diff
@@ -960,8 +949,8 @@ class _CallStatementBase():
     """ Type definitions for :any:`CallStatement` node type. """
 
     name: Expression
-    arguments: Optional[Tuple[Expression, ...]] = None
-    kwarguments: Optional[Tuple[Tuple[str, Expression], ...]] = None
+    arguments: Optional[Tuple[Expression, ...]] = ()
+    kwarguments: Optional[Tuple[Tuple[str, Expression], ...]] = ()
```

Comment:
Is this a change in behaviour, i.e., `arguments` and `kwarguments` always a tuple now instead of `None`? (I'm not opposed, just trying to understand the impact)

Reply:
Internally, we would always force this into `()` in the pre-validator autocast mechanism. The reason I need to change it here is that the before-validator does not get invoked if the arg is omitted so the default is used directly. This should be agnostic to the previous behaviour. 🤞